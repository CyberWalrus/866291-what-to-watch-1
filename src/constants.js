const GENRE_DEFOULT = `All genres`;
const NUMBER_FILM = 20;
const NUMBER_FILM_GENRE = 4;
const SERVER_URL = `https://es31-server.appspot.com/wtw`;
const SERVER_URL_USER = `https://es31-server.appspot.com/`;
const LINK_STYLE = `inherit`;
const MAIN_FILM_ID = 1;
const PADDING_VIDEO = 25;
const REVIEW_MESSAGE = `OK`;

const ratingRadioValues = [`1`, `2`, `3`, `4`, `5`];

const RatingName = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};
const RatingInterval = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERY_GOOD: 10,
  AWESOME: 10
};
const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
const BodyOverflow = {
  VISIBLE: `visible`,
  HIDDEN: `hidden`
};
const FavoriteStatus = {
  ADD: 1,
  REMOVE: 0
};
const TextLength = {
  MIN: 50,
  MAX: 400
};
const RatingValue = {
  MIN: 1,
  MAX: 5
};

const DisabledStyle = {
  opacity: `0.4`,
  border: `inherit`,
  cursor: `not-allowed`
};
const OptionsVideoMin = {
  width: `280`,
  height: `175`,
  isMuted: true,
  isLoop: true,
  isControls: false
};
const OptionsVideoFull = {
  width: `100%`,
  height: `100%`,
  isMuted: false,
  isLoop: false,
  isControls: false
};

export {
  GENRE_DEFOULT,
  NUMBER_FILM,
  SERVER_URL,
  SERVER_URL_USER,
  NUMBER_FILM_GENRE,
  LINK_STYLE,
  PADDING_VIDEO,
  REVIEW_MESSAGE,
  MAIN_FILM_ID,
  ratingRadioValues,
  RatingName,
  RatingInterval,
  FilmRoute,
  OptionsVideoMin,
  OptionsVideoFull,
  BodyOverflow,
  FavoriteStatus,
  DisabledStyle,
  TextLength,
  RatingValue
};
