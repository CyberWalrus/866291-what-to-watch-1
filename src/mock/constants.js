const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
const GENRE_DEFOULT = `All genres`;
const NUMBER_FILM = 20;

export {FilmRoute, GENRE_DEFOULT, NUMBER_FILM};

