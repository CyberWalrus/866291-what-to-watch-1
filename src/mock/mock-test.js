const FILMS = [
  {
    bgColor: `#977461`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
    director: `Martin Scorsese`,
    genre: `Thriller`,
    id: 1,
    isFavorite: false,
    pageUrl: `Shutter Island`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: `8,1`,
    ratingLevel: `Very good`,
    released: 2010,
    runTime: 138,
    scoresCount: 1002557,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
    starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
    title: `Shutter Island`
  },
  {
    bgColor: `#F1E9CE`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    director: `Justin Kurzel`,
    genre: `Drama`,
    id: 2,
    isFavorite: false,
    pageUrl: `Macbeth`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: `6,6`,
    ratingLevel: `Good`,
    released: 2015,
    runTime: 113,
    scoresCount: 48798,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Macbeth.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg`,
    starrings: [(`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`)],
    title: `Macbeth`
  }
];
const FILM = {
  bgColor: `#977461`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
  director: `Martin Scorsese`,
  genre: `Thriller`,
  id: 1,
  isFavorite: false,
  pageUrl: `Shutter Island`,
  srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: `8,1`,
  ratingLevel: `Very good`,
  released: 2010,
  runTime: 138,
  scoresCount: 1002557,
  srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
  srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
  srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
  starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
  title: `Shutter Island`
};
const REVIEWS = [
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  },
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  }
];

const GENRES = [
  `All genres`,
  `Thriller`,
  `Drama`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Action`,
  `Fantasy`
];
export {FILMS, FILM, GENRES, REVIEWS};
