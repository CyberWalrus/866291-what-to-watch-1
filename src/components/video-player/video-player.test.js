import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player.jsx";
import {FILM} from "../../mock/mock-test.js";
import {OptionsVideoFull} from "../../constants.js";

it(`VideoPlayer correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <VideoPlayer
          videoSrc={FILM.srcVideo}
          posterSrc={FILM.srcPosterImage}
          options={OptionsVideoFull}
          isPlaying={false}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
