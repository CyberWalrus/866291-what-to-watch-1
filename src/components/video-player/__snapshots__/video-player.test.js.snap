// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer correctly renders after relaunch 1`] = `
<video
  className="player__video"
  controls={false}
  height="100%"
  loop={false}
  muted={false}
  poster="https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg"
  src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
  width="100%"
/>
`;
